awscli/customizations/generatecliskeleton.py

```diff
@@ -59,29 +66,66 @@ def _register_argument_action(self):
             'calling-command.*', self.generate_json_skeleton)
         super(GenerateCliSkeletonArgument, self)._register_argument_action()
 
+    def override_required_args(self, argument_table, args, **kwargs):
+        arg_name = '--' + self.name
+        if arg_name in args:
+            arg_location = args.index(arg_name)
+            try:
+                # If the value of --generate-cli-skeleton is ``output``,
+                # do not force required arguments to be optional as
+                # ``--generate-cli-skeleton output`` validates commands
+                # as well as print out the sample output.
+                if args[arg_location + 1] == 'output':
+                    return
+            except IndexError:
+                pass
+            super(GenerateCliSkeletonArgument, self).override_required_args(
+                argument_table, args, **kwargs)
+
     def generate_json_skeleton(self, call_parameters, parsed_args,
                                parsed_globals, **kwargs):
-
-        # Only perform the method if the ``--generate-cli-skeleton`` was
-        # included in the command line.
-        if getattr(parsed_args, 'generate_cli_skeleton', False):
-
-            # Obtain the model of the operation
+        if getattr(parsed_args, 'generate_cli_skeleton', None):
+            for_output = parsed_args.generate_cli_skeleton == 'output'
             operation_model = self._operation_model
 
-            # Generate the skeleton based on the ``input_shape``.
-            argument_generator = ArgumentGenerator()
-            operation_input_shape = operation_model.input_shape
-            # If the ``input_shape`` is ``None``, generate an empty
-            # dictionary.
-            if operation_input_shape is None:
-                skeleton = {}
+            if for_output:
+                service_name = operation_model.service_model.service_name
+                operation_name = operation_model.name
+                return StubbedCLIOperationCaller(self._session).invoke(
+                    service_name, operation_name, call_parameters,
+                    parsed_globals)
             else:
-                skeleton = argument_generator.generate_skeleton(
-                    operation_input_shape)
-
-            # Write the generated skeleton to standard output.
-            sys.stdout.write(json.dumps(skeleton, indent=4))
-            sys.stdout.write('\n')
-            # This is the return code
-            return 0
+                argument_generator = ArgumentGenerator()
```

Comment:
Looks like my original comment was lost somewhere, but just to capture:

I think it would be nice if both cases just delegated out to a class to handle the main logic.  In the `for_output` case, the actual behavior is delegated to another class, but the else clause has the logic all inline.


```
if for_output:
    set_stuff_up()
    RealObject.handle_it()
else:
    set_stuff_up()
    RealObject2.handle_it()